R/checks.R

```diff
@@ -789,4 +789,45 @@ check_fracwith_diswith <- function(form, rep) {
     filter(ptstatus == 1 & (num_of_fractures > 1 |
                               location_of_fracture != location_of_dislocation))
   return(problems)
+}
+
+#' Check that the time from injury to def. stabilization should be within
+#'  1 hour of the time from injury to abx administration
+#' 
+#' @param form dataframe containing ptstatus, form 4.1, and form5.3
+#' @return a dataframe containing problematic entries with relevant columns
+#' @import tidyverse
+#' @export 
+check_locof1stadmin_pritosur <- function(form) {
+  hrswithin <- 1
+  problems <- form %>% 
+    transmute(
+      region, site, studyid, ptinit, ptstatus, ihhrs, ishrs_1, iahrs,
+      itoshrs = ihhrs + ishrs_1, 
+      comment = "The time from injury to def. stabilization should be within 1 hour of the time from injury to abx administration") %>% 
+    filter(ptstatus == 1 & !(abs(itoshrs - iahrs) <= hrswithin | ihhrs < 0 | ishrs_1 < 0 | iahrs < 0))
+  return(problems)
+}
+
+#' Check that the time from injury to abx administration should be no more than
+#'  12 hours from the time from injury to def. stabilization
+#' 
+#' @param form dataframe containing ptstatus, form 4.1, and form5.3
+#' @return a dataframe containing problematic entries with relevant columns
+#' @import tidyverse
+#' @export 
+check_locof1stadmin_ope <- function(form) {
+  hrsmore <- 12
+  problems <- form %>% 
+    transmute(
+      region, site, studyid, ptinit, ptstatus, 
+      ihhrs = coalesce(pull(form, ihhrs), 0), ihdays = coalesce(pull(form, ihdays), 0),
```

Comment:
Never change the values of existing columns. If you work on the data, create a new column instead.

Reply:
Also you don't have to pull. Using a col name in tidyverse function automatically means pull out that column as a vector.